projects/bms_carrier/test/test_passive_balance.c

```diff
@@ -0,0 +1,190 @@
+// Test sequence for passive balancing module
+#include "passive_balance.h"
+
+#include "bms_events.h"
+#include "delay.h"
+#include "interrupt.h"
+#include "ltc_afe.h"
+#include "ms_test_helpers.h"
+#include "stdio.h"
+#include "string.h"
+
+#include "log.h"
+
+// Just placeholders for afe settings from test_ltc_afe.c
+#define TEST_LTC_AFE_ADC_MODE LTC_AFE_ADC_MODE_7KHZ
+#define TEST_LTC_AFE_SPI_PORT SPI_PORT_1
+#define TEST_LTC_AFE_SPI_BAUDRATE 750000
+#define TEST_LTC_AFE_SPI_MOSI \
+  { .port = GPIO_PORT_A, .pin = 7 }
+#define TEST_LTC_AFE_SPI_MISO \
+  { .port = GPIO_PORT_A, .pin = 6 }
+#define TEST_LTC_AFE_SPI_SCLK \
+  { .port = GPIO_PORT_A, .pin = 5 }
+#define TEST_LTC_AFE_SPI_CS \
+  { .port = GPIO_PORT_A, .pin = 4 }
+
+// Number of cell voltages to be directly modified during testing
+static const uint8_t NUM_TEST_VOLTAGES = 4;
+
+static LtcAfeSettings s_test_afe_settings = {
+  .num_cells = LTC_AFE_MAX_CELLS,
```

Comment:
Probably should use the number of cells that we'd actually use. Internally we have an oversized array so using the max could mask invalid access bugs technically

Reply:
You know off the top of your head how many we're using?  I can ask hardware if not

Reply:
Checkout `cell_sense.h` I believe that's aligned with hardware